PythonPonies/bangazon-storefront
# first iteration of navigation bar

# Description
This PR pushes up a navigation bar that shows how many products a user has while they are visiting the home page.

## Number of Fixes
None, this is new code.

## Related Ticket(s)
#4 

## Problem to Solve
We need navigation to navigate between available options in the app. We also need to see how many products are in the cart. Note, this iteration only works from the homepage as far as showing how many products are on an order for the active user.

## Proposed Changes
This PR shows the first iteration of the navigation

## Expected Behavior
This iteration only works from the homepage, but the navbar does allow you to navigate between pages.

## Steps to Test Solution

1. fetch this branch
1. run `python manage.py test banagazon_storefront`
1. All tests should pass except 1 (test_get_all_product_types)
1. Try navigating the app. Everything but adding a product shows you content.
1. If you add a product via the admin for the currently logged in user, you'll see the shopping chart number change.

## Testing

[x] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[x] I added documentation for any new classes/methods

## Deploy Notes
none
